DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h

```diff
@@ -269,14 +302,19 @@ class TTTrack_TrackWord {
     return (up - bins.begin() - 1);
   }
 
-  double unpackSignedValue(unsigned int bits, unsigned int nBits, double lsb) const {
-    int isign = 1;
-    unsigned int digitized_maximum = (1 << nBits) - 1;
-    if (bits & (1 << (nBits - 1))) {  // check the sign
-      isign = -1;
-      bits = (1 << (nBits + 1)) - bits;  // if negative, flip everything for two's complement encoding
+  double undigitizeSignedValue(unsigned int twosValue, unsigned int nBits, double lsb) const {
+    // Check that none of the bits above the nBits-1 bit, in a range of [0, nBits-1], are set.
+    // This makes sure that it isn't possible for the value represented by `twosValue` to be
+    //  any bigger than ((1 << nBits) - 1).
+    assert((twosValue >> nBits) == 0);
+
+    // Convert from twos compliment to floating point values
```

Comment:
I'm not sure what a non-twos-complement digitized number is called. But I'd suggest comment at L311 should be "Convert from twos compliment to normal digitized value", and then before L317, you need comment "Convert to floating point value".

Reply:
Fixed